app/models/project.rb

```diff
@@ -99,12 +120,26 @@ def self.can_edit?(obj, user)
 
     group_ids = user.user_groups.map(&:id)
     obj.projects.each do |project|
+      next if project.open_membership
       return true if group_ids.member?(project.user_group_id) ||
                      group_ids.member?(project.admin_group_id)
     end
     false
   end
 
+  # Check if this user is an admin for a project that includes
+  # this observation.
+  def self.admin_power?(observation, user)
+    return false unless user
+    return false if observation.projects.empty?
+
+    group_ids = user.user_groups.map(&:id)
```

Comment:
Don't we get `User#user_group_ids` for free with the association?  Might need to be `user.user_group_ids.to_a`, though, not sure.

Reply:
Good catch.  I was just copying what was in self.can_edit? above, but what you suggest works.  It makes the same DB queries, but it is already defined.  Not clear why it was that way.